File: client/src/api/types/response/GlobalNotificationsSettings.ts
import {NotificationsSettings} from "./NotificationsSettings";
import {ChatNotificationsSettings} from "./ChatNotificationsSettings";

export interface GlobalNotificationsSettings {
	groupChats: NotificationsSettings,
	groupChatsExceptions: ChatNotificationsSettings[],
	dialogs: NotificationsSettings,
	dialogChatsExceptions: ChatNotificationsSettings[]
}

File: chat/src/main/kotlin/chatox/chat/model/GlobalNotificationsSettings.kt
package chatox.chat.model

interface GlobalNotificationSettings {
    val groupChats: NotificationsSettings
    val dialogs: NotificationsSettings

    companion object DEFAULT: GlobalNotificationSettings {
        override val groupChats: NotificationsSettings = NotificationsSettings(
                level = NotificationLevel.ALL_MESSAGES,
                sound = NotificationSound.HAPPY_POP_3
        )
        override val dialogs: NotificationsSettings = NotificationsSettings(
                level = NotificationLevel.ALL_MESSAGES,
                sound = NotificationSound.HAPPY_POP_3
        )
    }
}